orca-qos/src/main/kotlin/com/netflix/spinnaker/orca/qos/bufferstate/ActiveExecutionsBufferStateSupplier.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.orca.qos.bufferstate
+
+import com.netflix.spectator.api.Registry
+import com.netflix.spinnaker.orca.qos.BufferState
+import com.netflix.spinnaker.orca.qos.BufferState.ACTIVE
+import com.netflix.spinnaker.orca.qos.BufferState.INACTIVE
+import com.netflix.spinnaker.orca.qos.BufferStateSupplier
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Component
+
+/**
+ * Supplies the buffering state based on a threshold of system-wide active executions.
+ *
+ * As an immediate implementation, this supplier depends on [RedisActiveExecutionMonitor] to update the registry with
+ * system-wide active executions. The polling interval of that monitor will determine the resolution of the buffering
+ * state. By default, this is 60 seconds.
+ */
+@Component
+class ActiveExecutionsBufferStateSupplier(
+  @Value("\${qos.bufferingState.activeExecutionsThreshold:100}") private val threshold: Int,
+  private val registry: Registry
+) : BufferStateSupplier {
+
+  private val log = LoggerFactory.getLogger(ActiveExecutionsBufferStateSupplier::class.java)
+
+  private var state: BufferState = INACTIVE
+
+  private val bufferingId = registry.createId("qos.buffering")
+
+  @Scheduled(fixedDelayString = "\${pollers.qosUpdateStateIntervalMs:5000}")
```

Comment:
Not sure this `pollers.qosUpdateStateIntervalMs` pattern fits the existing structure of `pollers`

Perhaps

```
pollers:
  qos:
    updateStateIntervalMs: 5000
```

We have enough spare threads in our `orca` threadpool for these new `@Scheduled` right? I think we do but it's worth being conscious of what that max number of threads is.

Reply:
By default the `taskScheduler` was a single threaded pool but it's been increased -- from this metric it looks like we're 10 per instance.

These pollers are running per-instance (right?) so we just need to be conscious that we don't have 10 `@Scheduled` as we run the risk that something goes sideways and execution gets all out of wack (it's happened with the default single threaded pool).